crates/aptos/src/move_tool/mod.rs

```diff
@@ -1727,30 +1758,58 @@ async fn is_staging_area_empty(
     txn_options: &TransactionOptions,
     large_packages_module_address: AccountAddress,
 ) -> CliTypedResult<bool> {
-    let client = txn_options.rest_client()?;
-
-    let (_, account_address) = txn_options.get_public_key_and_address()?;
-    let staging_area_response = client
-        .get_account_resource(
-            account_address,
-            &format!(
-                "{}::large_packages::StagingArea",
-                large_packages_module_address
-            ),
-        )
-        .await;
+    match &txn_options.session {
+        None => {
+            let client = txn_options.rest_client()?;
+
+            let (_, account_address) = txn_options.get_public_key_and_address()?;
+            let staging_area_response = client
+                .get_account_resource(
+                    account_address,
+                    &format!(
+                        "{}::large_packages::StagingArea",
+                        large_packages_module_address
+                    ),
+                )
+                .await;
 
-    match staging_area_response {
-        Ok(response) => match response.into_inner() {
-            Some(_) => Ok(false), // StagingArea is not empty
-            None => Ok(true),     // TODO: determine which case this is
+            match staging_area_response {
+                Ok(response) => match response.into_inner() {
+                    Some(_) => Ok(false), // StagingArea is not empty
+                    None => Ok(true),     // TODO: determine which case this is
+                },
+                Err(RestError::Api(aptos_error_response))
+                    if aptos_error_response.error.error_code
+                        == AptosErrorCode::ResourceNotFound =>
+                {
+                    Ok(true) // The resource doesn't exist
+                },
+                Err(rest_err) => Err(CliError::from(rest_err)),
+            }
         },
-        Err(RestError::Api(aptos_error_response))
-            if aptos_error_response.error.error_code == AptosErrorCode::ResourceNotFound =>
-        {
-            Ok(true) // The resource doesn't exist
+        Some(session_path) => {
+            let sess = Session::load(session_path)?;
+
+            let (_, account_address) = txn_options.get_public_key_and_address()?;
+
+            let staging_area_bytes = sess
+                .state_store()
+                .get_resource_bytes(
+                    &StateKey::resource(&account_address, &StructTag {
```

Comment:
nit: I think you can use `FromStr` for tag, so re-use `format!("{}::large_packages::StagingArea", large_packages_module_address)` from above case.